### PR TITLE
feat: add typosquatting analyzer for go modules

### PR DESCRIPTION
**Pull request by bthuilot**
## Changes
- Adds `typosquatting` analyzer for go modules
- Adds `top_go_packages.json` which contains top ~3000 go modules
  - List was gathered by searching for go.mod files on GitHub with more than 50 stars
  - Parsed the go.mod files and counted the amount of each pacakge used, then took all packages used in more than 10 repos

### CODE REVIEW

**Comment by sobregosodd** [comment]
> which contains top 500 go modules

@bthuilot mind that we are using about 80k packages in other ecosystems for this detection

File: guarddog/analyzer/metadata/go/typosquatting.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    Attributes:
+        popular_packages (set): set of top 500 most popular Go packages,
+          as determined by count of references across top starred repositories
+    """
+
+    def _get_top_packages(self) -> set:
+
+        # popular_packages_url = (
+        #     ""
+        # )
```
Comment (juliendoutre) [inline guarddog/analyzer/metadata/go/typosquatting.py:22]: nit: dead code?

File: guarddog/analyzer/metadata/go/typosquatting.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            #
+            # if datetime.now() - update_time <= timedelta(days=30):
+            with open(top_packages_path, "r") as top_packages_file:
+                top_packages_information = json.load(top_packages_file)
+
+        # if top_packages_information is None:
+        #     response = requests.get(popular_packages_url).json()
+        #     top_packages_information = list([i["name"] for i in response[0:8000]])
+        #     with open(top_packages_path, "w+") as f:
+        #         json.dump(top_packages_information, f, ensure_ascii=False, indent=4)
```
Comment (juliendoutre) [inline guarddog/analyzer/metadata/go/typosquatting.py:47]: dead code?

File: guarddog/analyzer/metadata/go/typosquatting.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            )
+
+        top_packages_path = os.path.join(resources_dir, top_packages_filename)
+
+        top_packages_information = None
+
+        if top_packages_filename in os.listdir(resources_dir):
+            # update_time = datetime.fromtimestamp(os.path.getmtime(top_packages_path))
+            #
+            # if datetime.now() - update_time <= timedelta(days=30):
```
Comment (juliendoutre) [inline guarddog/analyzer/metadata/go/typosquatting.py:39]: dead code?

**Review by juliendoutre — COMMENTED** [review]
Looks good to me but noticed a few commented code sections which may not be intentional?

**Review by sobregosodd — CHANGES_REQUESTED** [review]

**Review by sobregosodd — APPROVED** [review]

**Pull request merged**